add missing 'lib' argument in guarded main_ calls, fixes "no matching function for call" errors

author: Kenneth Hoste (HPC-UGent)

--- trilinos-12.12.1-Source/src/packages/muelu/test/maxwell/Maxwell3D.cpp.orig	2018-09-04 17:51:13.645528235 +0200
+++ trilinos-12.12.1-Source/src/packages/muelu/test/maxwell/Maxwell3D.cpp	2018-09-04 17:52:30.626525782 +0200
@@ -355,7 +355,7 @@
         typedef Kokkos::Compat::KokkosOpenMPWrapperNode Node;
 
 #  ifndef HAVE_MUELU_EXPLICIT_INSTANTIATION
-        return main_<double,int,int,Node>(clp, argc, argv);
+        return main_<double,int,int,Node>(clp, lib, argc, argv);
 #  else
 #    if   defined(HAVE_TPETRA_INST_DOUBLE) && defined(HAVE_TPETRA_INST_OPENMP) && defined(HAVE_TPETRA_INST_INT_INT)
         return main_<double,int,int,Node> (clp, lib, argc, argv);
@@ -375,7 +375,7 @@
         typedef Kokkos::Compat::KokkosCudaWrapperNode Node;
 
 #  ifndef HAVE_MUELU_EXPLICIT_INSTANTIATION
-        return main_<double,int,int,Node>(clp, argc, argv);
+        return main_<double,int,int,Node>(clp, lib, argc, argv);
 #  else
 #    if defined(HAVE_TPETRA_INST_DOUBLE) && defined(HAVE_TPETRA_INST_CUDA) && defined(HAVE_TPETRA_INST_INT_INT)
         return main_<double,int,int,Node> (clp, lib, argc, argv);
